rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py

```diff
@@ -0,0 +1,81 @@
+from panther_base_helpers import deep_get
+
+EC2_CRUD_ACTIONS = {
+    "AssociateIamInstanceProfile",
+    "AssociateInstanceEventWindow",
+    "BundleInstance",
+    "CancelSpotInstanceRequests",
+    "ConfirmProductInstance",
+    "CreateInstanceEventWindow",
+    "CreateInstanceExportTask",
+    "DeleteInstanceEventWindow",
+    "DeregisterInstanceEventNotificationAttributes",
+    "DisassociateIamInstanceProfile",
+    "DisassociateInstanceEventWindow",
+    "ImportInstance",
+    "ModifyInstanceAttribute",
+    "ModifyInstanceCapacityReservationAttributes",
+    "ModifyInstanceCreditSpecification",
+    "ModifyInstanceEventStartTime",
+    "ModifyInstanceEventWindow",
+    "ModifyInstanceMaintenanceOptions",
+    "ModifyInstanceMetadataOptions",
+    "ModifyInstancePlacement",
+    "MonitorInstances",
+    "RegisterInstanceEventNotificationAttributes",
+    "ReportInstanceStatus",
+    "RequestSpotInstances",
+    "ResetInstanceAttribute",
+    "RunInstances",
+    "RunScheduledInstances",
+    "StartInstances",
+    "StopInstances",
+    "TerminateInstances",
+    "UnmonitorInstances",
+}
+
+
+def rule(event):
+    # Disqualify any eventSource that is not ec2
+    if event.get("eventSource", "") != "ec2.amazonaws.com":
+        return False
+    # Disqualify AWS Service-Service operations, which can appear in a variety of forms
+    if (
+        # FYI there is a weird quirk in the sourceIPAddress field of CloudTrail
```

Comment:
@calkim-panther  - There's kind of a lot in this updated comment. I hope it's more clear than the previous version